### PR TITLE
Adds ExtensionsManager.lookupExtensionSettingsById

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Companion PR in the security plugin: https://github.com/opensearch-project/security/pull/2749

Adds a public method to lookup extension settings placed in `extensions.yml` by the extension unique id. There was a recent issue with extension handshake and the security plugin where the only available method exposed looks up initialized extensions which is not populated at the time of initial handshake: https://github.com/opensearch-project/security/issues/2747

This new method is utilized in the security plugin to ensure that the unique id exists in `extensions.yml` so that it can permit the transport requests that an extension uses to bootstrap

### Related Issues
Resolves https://github.com/opensearch-project/security/issues/2747

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/extensions/ExtensionsManager.java [hunk 1]
```diff
@@ -308,6 +321,7 @@ protected void discover() throws IOException {
                 throw new IOException("Could not read from extensions.yml", e);
             }
             for (Extension extension : extensions) {
+                extensionSettingsMap.put(extension.getUniqueId(), extension);
```
Comment (peternied) [inline server/src/main/java/org/opensearch/extensions/ExtensionsManager.java:324]: loadExtension(...) has logic to ignore / error check that might cause the lists to be inconsistent, I'd recommend moving this `put` into the end of the load extension method

Comment (cwperks) [reply server/src/main/java/org/opensearch/extensions/ExtensionsManager.java]: I moved this so that its next to the call to `extensionIdMap.put` now.

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
@cwperks Could you please investigate the test failure to see if this is a flaky test?
```
org.opensearch.indices.replication.SegmentReplicationIT.testReplicationPostDeleteAndForceMerge
```

**Review by DarshitChanpura — APPROVED** [review]
Nicely done. This will fix the TLS setup issue with extensions!

**Review by saratvemulapalli — APPROVED** [review]

**Comment by cwperks** [comment]
I'll open a manual backport for this

**Pull request merged**